src/Compilers/VisualBasic/Portable/Symbols/MethodSignatureComparer.vb

```diff
@@ -425,8 +426,10 @@ Done:
             type1 = SubstituteType(typeSubstitution1, type1)
             type2 = SubstituteType(typeSubstitution2, type2)
 
-            If Not type1.Type.IsSameTypeIgnoringCustomModifiers(type2.Type) Then
+            If Not type1.Type.IsSameType(type2.Type, TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds Or TypeCompareKind.IgnoreTupleNames) Then
                 Return SymbolComparisonResults.ReturnTypeMismatch
+            ElseIf Not type1.Type.IsSameType(type2.Type, TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds) Then
+                Return SymbolComparisonResults.TupleNamesMismatch
             ElseIf (comparisons And SymbolComparisonResults.CustomModifierMismatch) <> 0 AndAlso
                    (type1 <> type2) Then
```

Comment:
>type1 <> type2 [](start = 20, length = 14)

I think the code will be more robust long term if we would do comparison with explicit options here.